Reactivate commown module's CI

fcayre opened the pull request.

fcayre committed:
Reactivate commown module's CI

fcayre committed:
Update debian package list before installing

This should make sure the install does not fail because of an obsolete package
(because of a security update for instance).

fcayre committed:
Add website_sale_payment_slimpay PR in the test requirements

legalsylvain commented:
Je ne comprend pas l'erreur. 

Python-magic est bien installé. (Grace a l'override dans le setup.py de commown.

https://github.com/commown/commown-odoo-addons/runs/5476903763?check_suite_focus=true#step:5:97

Une idee ?

fcayre committed:
Install libmagic1 and librsvg2-bin debian packages

The former is required by the advanced "python-magic" implementation
(much better than the "magic" python package thanks to this lib).

fcayre commented:
@legalsylvain non, le code de setuptools-odoo semble pourtant clair à ce sujet : https://github.com/acsone/setuptools-odoo/blob/master/setuptools_odoo/core.py#L266

J'ai essayé de l'ajouter à test-requirements pour voir...

fcayre commented:
@legalsylvain ah compris : ImportError: failed to find libmagic.  Check your installation

Il faut installer le paquet debian libmagic d'abord... je fais la modif !

legalsylvain commented:
alors du coup, il faut peut-être le mettre dans le manifest du module commown ? ``external_dependencies/bin``, non ?

nota il manque l'installation de rsvg-convert j'ai l'impression aussi.

fcayre commented:
@legalsylvain je ne sais pas pour external dependencies, j'aurais tendance à penser que non. Il me semble que c'est plutôt fait pour vérifier la disponibilité de binaires exécutables et non pas de fichiers binaires en général (mais je ne suis pas sûr). Or il n'y a pas de binaire exécutable dans le paquet debian libmagic1.
Oui vu pour rsvg-convert, je vais regarder ma config salt pour lister de manière exhaustive les paquets debian pré-requis.

legalsylvain commented:
> @legalsylvain je ne sais pas pour external dependencies, j'aurais tendance à penser que non. Il me semble que c'est plutôt fait pour vérifier la disponibilité de binaires exécutables et non pas de fichiers binaires en général (mais je ne suis pas sûr). Or il n'y a pas de binaire exécutable dans le paquet debian libmagic1.

Oui tu as raison ! 

j'avais dans l'idée à vrai dire, d'améliorer la CI de l'OCA, pour auto-générer un ``debian-requirements.txt`` de la même façon que le ``requirements.txt`` pour les paquets python. Après, dans la CI, on aura plus qu'à faire un apt-get install du fichier ``debian-requirements.txt``, et ça serait fini. (plutôt que de les énumérer à la main).

Par contre, peut-être que ça va merder si ``libmagic`` n'est pas dans le path, parce qu'il n'y pas de binaire associé. je ne sais pas exactement comment le test est réalisé...

fcayre committed:
Fix redirect controller tests

fcayre committed:
Move MockedEmptySessionMixin to product_rental and make RentalSaleOrderTC inherit it

This should avoid request-related errors on tests running in an
environment where website_sale_affiliate is installed.

fcayre committed:
Fix custom_report tests to make them work with website_sale_affiliate installed

fcayre committed:
Fix product_rental tests

2 fixes:

- a bad interaction with contract_queue_job, installed in the CI

  The _recurring_create_invoice method does not return anything
  when this module is installed, even when configuring the queue
  module to sync mode, so we need to fetch the created invoices
  from the database.

- a change in the generated contracts and invoices (order was
  somehow reversed from 10.0 to 12.0). Current implementation
  still depends on this order, which is fragile and should be
  fixed ASAP

fcayre committed:
Move mocked session test mixin class setup into instance setup

Otherwise concrete test classes that implement it are not
initialized with their env when we try to modify it and
the class init crashes.

legalsylvain commented:
Victory !

fcayre commented:
\o/ juste avant le week-end c'est cool :)
